presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanChecker.java

```diff
@@ -74,6 +74,9 @@ public PlanChecker(FeaturesConfig featuresConfig, boolean forceSingleNode)
                         new VerifyProjectionLocality(),
                         new DynamicFiltersChecker(),
                         new WarnOnScanWithoutPartitionPredicate(featuresConfig))
+                .putAll(
+                        Stage.RUNTIME,
```

Comment:
We shouldn't have a seperate runtime planchecker, since the runtime plan has the same requirements as the "final" plan. Instead, we should be running the planchecker for the final stage after doing runtime optimization

Reply:
Discussed offline. We do not want to modify any existing checks in the FINAL and FRAGMENT stages, but there are problems for using each of the two stage to carry out runtime checking (e.g. FINAL is targeting a whole plan, FRAGMENT is generic whereas the streaming property check is specific to fragments with their exchange sources materialized). So decide to add a RUNTIME stage and copy all necessary checks into this stage. Future runtime checks around exchange materialized stages will also go into this new category.

Reply:
sorry, been thinking about this again.  is there a way to make this check in the "fragment" section, but for it to only run if the plan has a tablescan stage.  Because the check isn't specific to anything that happens at runtime, it's just that runtime optimization happens to only involve certain kinds of fragments

Reply:
Sure, totally make sense.